content/en/docs/Advanced/configuring_partitions.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Configuring partitions"
+linkTitle: "Configuring partitions"
+weight: 1
+description: >
+---
+
+{{% alert title="Note" color="warning" %}}
+
+This feature will be available in Kairos version `2.4.0` and in all future releases.
+
+{{% /alert %}}
+
+
+Kairos configuration mechanism for partitions is based on the [cloud-config]({{< relref "../Reference/configuration" >}}) file 
+given during installation to override the default values set by the installer.
+
+We allow certain flexibility in the sizes and filesystems used for the default install and allow to create extra partitions as well.
+
+For example, the following cloud-config will make the `oem` partition have a size of 512Mb and an `ext4` filesystem, 
+recovery with a size of 10000Mb and a `ext4` filesystem, while leaving the rest of the partitions to their default sizes and filesystems. 
+
+```yaml
+#cloud-config
+
+install:
+  device: "/dev/sda"
+  auto: true
+  partitions:
+    oem:
+      size: 512
+      fs: ext4
+    recovery:
+      size: 10000
+      fs: ext4
+```
+
+
+
+And the following config will leave the default partitions as is, but create 2 new extra partitions with the given sizes, filesystems and labels:
+
+```yaml
+#cloud-config
+
+install:
+  device: "/dev/sda"
+  auto: true
+  extra-partitions:
+    - name: first_partition
+      size: 512
+      fs: ext3
+    - name: second_partition
+      size: 100
+      fs: ext2
+      label: PARTITION_TWO
+```
+
+Only one partition can expand to the rest of the disk. 
+Either persistent or one of the extra-partitions. 
+In case you want the latter, you need to specify the size of persistent to a fixed value.
+
+An example of this would be as follows:
+
+```yaml
+#cloud-config
+
+install:
+  device: "/dev/sda"
+  auto: true
+  partitions:
+    persistent:
+      size: 500
+  extra-partitions:
+    - name: big_partition
+      size: 0
+      fs: ext3
+```
+
+
+Note that there are some caveats in the `extra partitions` setup:
```

Comment:
Shouldn't we also mention that if they try to override the labels of the default partitions, these will be ignored? (as per the PR here: https://github.com/kairos-io/kairos-agent/pull/108/files)

Reply:
Not really no, the label stuff is for the system partitions and its mentioned above that we only support fs and size overrides.

Reply:
I dont want to give them ideas that they can indeed override the labels lol, security through obscurity